### ISSUE TITLE
config.cmd throws "dotnet has stopped working"

### ISSUE THREAD

**Issue by felipeoriani** [Microsoft/vsts-agent#1127]
We have a VSTS Agent running on our environment to make a continuous integration. It was working fine on the last months. 

Today, Windows Server 2012 R2 performed a `blue screen` and now it is not working anymore. The Service is running on the windows `services` list as usual but on the `VSTS` online it shows that the agent is `offline`. So, we decide to try to reconfigure the agent and when we try to run the command:

    PS> .\config.cmd remove

on the poweshell, we got the error message:

[![enter image description here][1]][1]

I am not sure if it is a Windows problem or the an agent setup or even a `dotnet` setup. Any command I try to run such as:

     PS> .\config.cmd help
     PS> .\run.cmd

I got this same message. I have tried all these commands and without success.

How can I solve this problem? It was working fine but not it just stopped because of a `blue screen`.

  [1]: https://i.stack.imgur.com/oU6V0.png

## Agent version and platform
Agent 2.119.1. 

## VSTS type and version
On-Prem VSTS 2017

If VisualStudio.com, what is your account name? http://gantc.visualstudio.com

**Comment by TingluoHuang** [comment]
@felipeoriani can you take a process dump and share with me?

**Comment by felipeoriani** [comment]
Yes @TingluoHuang, I take a process dump and the file is 25MB. I will try to upload here.
[AgentService-Process-Dump.zip](https://github.com/Microsoft/vsts-agent/files/1209365/AgentService-Process-Dump.zip)

**Comment by TingluoHuang** [comment]
@felipeoriani take a dump for `agent.listener.exe` not the agentservice.exe. :)

**Comment by felipeoriani** [comment]
@TingluoHuang the problem is that the agent.listener.exe is not starting. If I look at the EventViewer, it logs a message the following messages:

First, it tries to start:


[![enter image description here][1]][1]

And then:

[![enter image description here][2]][2]

And then:

[![enter image description here][3]][3]

And finally, it does not work:

[![enter image description here][4]][4]


If I try any command to remove the service, it show me the error message on the first post.

Is there something I can do? :(

  [1]: http://i.imgur.com/A7nMM0a.png
  [2]: http://i.imgur.com/CllgGyg.png
  [3]: http://i.imgur.com/0Asxu10.png
  [4]: http://i.imgur.com/YSPgLdc.png

**Comment by TingluoHuang** [comment]
@felipeoriani `c0000005` means memory access deny which is a OS level failure. To get the dump you can call `run.cmd` under agent root and take dump when it crash.

**Comment by felipeoriani** [comment]
@TingluoHuang when I run the run.cmd from the PowerShell, I got the same error on the first post. "dotnet has stopped working". Now I run it and look at the Task Manager and a new process called "dotnet" starts. I dump it and it generates a file called Agent.Listener.DMP. I have zipped it and attach here.

I hope it is fine to see in more details. 👍 

[Agent.Listener.DMP.zip](https://github.com/Microsoft/vsts-agent/files/1209470/Agent.Listener.DMP.zip)

**Comment by TingluoHuang** [comment]
@felipeoriani have you try reboot? :)

**Comment by felipeoriani** [comment]
Yes, I have tried to reboot, I also run the [.Net Framework Setup Verification Tool ](https://blogs.msdn.microsoft.com/astebner/2008/10/13/net-framework-setup-verification-tool-users-guide/) and all versions are ok. Visual Studio 2015 Community (as a requirement for the agent) is on the machine and it is running fine as well.

In the VSTS online the Agent is offline. The Service is running but does not start properly as I show you on the images above. 

Is there any information on the dump file I send you? Or anything else I can try?

**Comment by TingluoHuang** [comment]
@felipeoriani what's your `blue screen` about?

**Comment by felipeoriani** [comment]
@TingluoHuang I look at the EventViewer and I think I have some useful information. Look logs bellow:


Information: Windows Report Error
```

Fault bucket , type 0
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: powershell.exe
P2: 6.3.9600.17415
P3: 54504e09
P4: System.Management.Automation.ni.dll
P5: 6.3.9600.17400
P6: 543c535c
P7: c0000005
P8: 00000000006f06ef
P9: 
P10: 

Attached files:
C:\Windows\ServiceProfiles\NetworkService\AppData\Local\Temp\WER77DB.tmp.appcompat.txt
C:\Windows\ServiceProfiles\NetworkService\AppData\Local\Temp\WER7FBC.tmp.WERInternalMetadata.xml
C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_powershell.exe_e9ceacfe6e8888995434af6d577f75a20264c3b_305feb35_cab_0b598354\memory.hdmp
C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_powershell.exe_e9ceacfe6e8888995434af6d577f75a20264c3b_305feb35_cab_0b598354\triagedump.dmp

These files may be available here:
C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_powershell.exe_e9ceacfe6e8888995434af6d577f75a20264c3b_305feb35_cab_0b598354

Analysis symbol: 
Rechecking for solution: 0
Report Id: 62e8e4c4-7c49-11e7-80d1-00155d011305
Report Status: 0
Hashed bucket: 
```

Information: VstsAgentService
```
Connecting to the server.
```

Information: VstsAgentService
```
2017-08-08 14:54:27Z: Listening for Jobs
```

Information: Windows Report Error
```
Fault bucket , type 0
Event Name: BlueScreen
Response: Not available
Cab Id: 0

Problem signature:
P1: 1000007e
P2: ffffffffc0000005
P3: fffff802364651c6
P4: ffffd00074fb97e8
P5: ffffd00074fb8ff0
P6: 6_3_9600
P7: 0_0
P8: 272_3
P9: 
P10: 

Attached files:
C:\Windows\Minidump\080817-24421-01.dmp
C:\Users\Administrator\AppData\Local\Temp\WER-139875-0.sysdata.xml

These files may be available here:
C:\Users\Administrator\AppData\Local\Microsoft\Windows\WER\ReportArchive\Kernel_1000007e_b5881a189b502b40a37414914a10472c76587b2b_00000000_cab_063a5673

Analysis symbol: 
Rechecking for solution: 0
Report Id: 080817-24421-01
Report Status: 2048
Hashed bucket: 
```

Information: Windows Report Error
```
Fault bucket , type 0
Event Name: BlueScreen
Response: Not available
Cab Id: 0

Problem signature:
P1: 18
P2: 0
P3: ffffe0018a68b4d0
P4: 10
P5: f000000000002b73
P6: 6_3_9600
P7: 0_0
P8: 272_3
P9: 
P10: 

Attached files:
C:\Windows\Minidump\080717-23750-01.dmp
C:\Users\Administrator\AppData\Local\Temp\WER-139875-0.sysdata.xml
C:\Users\Administrator\AppData\Local\Temp\WER569D.tmp.WERInternalMetadata.xml

These files may be available here:
C:\Users\Administrator\AppData\Local\Microsoft\Windows\WER\ReportQueue\Kernel_18_bc4f6fa5b0a0bbddd6fc9977dd234c34b27551_00000000_cab_063a5692

Analysis symbol: 
Rechecking for solution: 0
Report Id: 080717-23750-01
Report Status: 4
Hashed bucket: 
```

**Comment by TingluoHuang** [comment]
@felipeoriani i think something messed up with you both, since Powershell also crashed with `C000005`.

**Comment by bryanmacfarlane** [comment]
Yeah, this is starting to look like an OS / Machine issue if it's blue screening and powershell are crashing.

**Comment by felipeoriani** [comment]
That's sad. Maybe do I have to reinstall windows on this environment? Or does it still have something I can do?

**Comment by TingluoHuang** [comment]
@felipeoriani maybe check what changed in your OS, like windows updates or hardware changes, and undo them?

**Comment by felipeoriani** [comment]
@TingluoHuang no, the updates were disabled and the hardware didn't change. :(

**Comment by bryanmacfarlane** [comment]
@felipeoriani - after that, you might need to follow up with usually OS support channels.
